code/entities/chair/Armrest.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Sandbox;
+
+namespace Cinema;
+
+public partial class Armrest : BaseNetworkable
+{
+    public enum States : int
+    {
+        Lowered,
+        Raising,
+        Raised,
+        Lowering
+    }
+
+    public enum Sides : int
+    {
+        Left,
+        Right
+    }
+
+    [Net]
+    public CinemaChair Chair { get; set; }
+    [Net]
+    public ModelEntity Cupholded { get; set; }
+    [Net]
+    public States State { get; set; } = States.Lowered;
```

Comment:
This might not happen, but I have concerns that the `State` variable could easily get out of sync.

It seems that `State` is really dependent on whether or not the armrest anim parameter `ArmrestToggleParameter` is true/false.

Is calling `GetAnimParameterBool` very expensive? We could just do
```
public States State => (States)Chair.GetAnimParameterBool(ArmrestToggleParameter)
```

Saves us the concern of tracking state, and simplifies the `HandleAnimTag` code.

Reply:
I removed the intermediate states of Armrest.States, so now the only remaining states are Raised and Lowered, which correspond roughly to the aforementioned anim parameter. Now it's possible to set `HeldEntity` while the arm is still being lowered, which is probably fine.